doc/source/whatsnew/v2.0.0.rst

```diff
@@ -1016,6 +1016,7 @@ Numeric
 - Bug in DataFrame reduction methods (e.g. :meth:`DataFrame.sum`) with object dtype, ``axis=1`` and ``numeric_only=False`` would not be coerced to float (:issue:`49551`)
 - Bug in :meth:`DataFrame.sem` and :meth:`Series.sem` where an erroneous ``TypeError`` would always raise when using data backed by an :class:`ArrowDtype` (:issue:`49759`)
 - Bug in :meth:`Series.__add__` casting to object for list and masked :class:`Series` (:issue:`22962`)
+- Bug in :meth:`query` with ``engine="numexpr"`` and column names are ``min`` or ``max``, does not work correctly (:issue:`50937`)
```

Comment:
```suggestion
- Bug in :meth:`query` with ``engine="numexpr"`` and column names are ``min`` or ``max`` would raise a ``TypeError`` (:issue:`50937`)
```

Reply:
> 

Why did we replace it with a `TypeError `rather than the error it raises